src/tools/clippy/clippy_lints/src/returns.rs

```diff
@@ -212,14 +212,6 @@ fn check_final_expr<'tcx>(
                     check_final_expr(cx, arm.body, Some(arm.body.span), RetReplacement::Block);
                 }
             },
-            MatchSource::IfLetDesugar {
```

Comment:
Why can we just stop handling this case?

Reply:
Same as https://github.com/rust-lang/rust/pull/80357#discussion_r624517410.
Added a `needless_return.rs` test to show that the lint is working as expected -> https://github.com/rust-lang/rust/pull/80357/files#diff-324e8719adb56e60d34436af7504ccdcd23b9acc31d22f8f527810a40f0b8b81